plugins/in_splunk/splunk_prot.c

```diff
@@ -447,14 +497,16 @@ static int validate_auth_header(struct flb_splunk *ctx, struct mk_http_request *
     }
 
     if (auth_header != NULL && auth_header->val.len > 0) {
-        if (strncmp(ctx->auth_header,
-                    auth_header->val.data,
-                    strlen(ctx->auth_header)) == 0) {
-            return SPLUNK_AUTH_SUCCESS;
-        }
-        else {
-            return SPLUNK_AUTH_UNAUTHORIZED;
+        mk_list_foreach_safe(head, tmp, &ctx->auth_tokens) {
```

Comment:
if the list entry won't be deleted, you can use `mk_list_foreach()` , the `_safe_` version aims to provide an extra pointer to deal with cases where entries/nodes are being removed.